src/components/study/study-code-details.tsx

```diff
@@ -0,0 +1,36 @@
+import { use } from 'react'
+import { Divider, Stack, Text } from '@mantine/core'
+import { AlertNotFound } from '@/components/errors'
+import { getMemberIdFromIdentifierAction } from '@/server/actions/user.actions'
+import { getStudyAction } from '@/server/actions/study.actions'
+import { StudyJobFiles } from '@/app/member/[memberIdentifier]/study/[studyIdentifier]/review/study-job-files'
+import { latestJobForStudyAction } from '@/server/actions/study-job.actions'
+
+export function StudyCodeDetails(props: {
+    params: {
+        memberIdentifier: string
+        studyIdentifier: string
+    }
+}) {
+    const { memberIdentifier, studyIdentifier } = props.params
+
+    const member = use(getMemberIdFromIdentifierAction(memberIdentifier))
+    if (!member) {
+        return <AlertNotFound title="Member was not found" message="no such member exists" />
+    }
+
+    const study = use(getStudyAction(studyIdentifier))
+    if (!study) {
+        return <AlertNotFound title="Study was not found" message="no such study exists" />
+    }
```

Comment:
since StudyCodeDetails is called from other pages which should also be checking  for member & study existence, why do we need the checks and alerts?  Why don't we have the pages pass in the member & study for this to display?